src/tls/mod.rs

```diff
@@ -0,0 +1,575 @@
+            easy.issuer_cert_blob(cert.pem.as_bytes())?;
+        }
+
+        if let Some(path) = self.issuer_cert_path.as_ref() {
+            easy.issuer_cert(path)?;
+        }
+
+        if let Some(identity) = self.identity.as_ref() {
+            identity.set_opt(easy)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl SetOptProxy for TlsConfig {
+    fn set_opt_proxy<H>(&self, easy: &mut Easy2<H>) -> Result<(), curl::Error> {
+        if let Some(ciphers) = self.ciphers.as_ref() {
+            easy.proxy_ssl_cipher_list(ciphers)?;
+        }
+
+        easy.proxy_ssl_options(&self.curl_flags)?;
+        easy.proxy_ssl_verify_peer(self.danger_accept_invalid_certs)?;
+        easy.proxy_ssl_verify_host(self.danger_accept_invalid_hosts)?;
+
+        easy.proxy_ssl_min_max_version(
+            self.min_version.unwrap_or(SslVersion::Default),
+            self.max_version.unwrap_or(SslVersion::Default),
+        )?;
+
+        self.root_cert_store.set_opt_proxy(easy)?;
+
+        if let Some(cert) = self.issuer_cert.as_ref() {
+            easy.proxy_issuer_cert_blob(cert.pem.as_bytes())?;
+        }
+
+        if let Some(path) = self.issuer_cert_path.as_ref() {
+            easy.proxy_issuer_cert(path)?;
+        }
+
+        if let Some(identity) = self.identity.as_ref() {
+            identity.set_opt_proxy(easy)?;
+        }
+
+        Ok(())
+    }
+}
+
+/// Possible SSL/TLS versions that can be used.
+///
+/// Not all versions are supported by all TLS backends, and can vary depending
+/// how a user's system is configured. Requesting or requiring a version that
+/// the TLS backend does not allow or support will result in a runtime error.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum ProtocolVersion {
+    /// SSL v2. Support for this version is disabled in the default build
+    /// configuration and not even available any more in most TLS backends due
+    /// to being considered insecure.
+    Sslv2,
+
+    /// SSL v3. Support for this version is disabled in the default build
+    /// configuration and not even available any more in most TLS backends due
+    /// to being considered insecure.
+    Sslv3,
+
+    /// TLS v1.0. Considered insecure.
+    Tlsv10,
+
+    /// TLS v1.1. Considered insecure.
+    Tlsv11,
+
+    /// TLS v1.2
+    Tlsv12,
+
+    /// TLS v1.3
+    Tlsv13,
+}
+
+impl ProtocolVersion {
+    const fn curl_version(&self) -> SslVersion {
+        match self {
+            Self::Sslv2 => SslVersion::Sslv2,
+            Self::Sslv3 => SslVersion::Sslv3,
+            Self::Tlsv10 => SslVersion::Tlsv10,
+            Self::Tlsv11 => SslVersion::Tlsv11,
+            Self::Tlsv12 => SslVersion::Tlsv12,
+            Self::Tlsv13 => SslVersion::Tlsv13,
+        }
+    }
+}
+
+/// An X.509 digital certificate.
+#[derive(Clone, Debug)]
+pub struct Certificate {
+    /// Curl prefers to work in the PEM format, so internally we do as well.
+    pem: String,
```

Comment:
I don't if it's better to store as `Vec<u8>`?

Reply:
Well internally this eventually gets passed as a curl option such as [`CURLOPT_CAINFO_BLOB`](https://curl.se/libcurl/c/CURLOPT_CAINFO_BLOB.html) which requires PEM format, and seeing as PEM is a textual format a `String` makes sense here.